### PR TITLE
DOC-1640: Swap to using GitHub Actions

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: DOC-1640

Description of Changes:
* Merge in all changes from `main`
* Setup GitHub actions to deploy the legacy jekyll site (see the last commit).
  * **Note:** I have **not** included the check links part of the staging build as that requires the `linkchecker` executable to be available. I'm not sure how to get that to work with GitHub actions and it didn't seem important given we're moving to a different docs system soon anyways.

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: .github/workflows/feature_5_legacy_docs.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: Develop > Legacy Tiny 5 Docs
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - '**'
+    branches:
+      - 'feature/5/DOC-1640'
```
Comment (lnewson) [inline .github/workflows/feature_5_legacy_docs.yml:9]: This was very much intentional, as we can't use `feature/*` or `hotfix/*` as then it'd run on any of our Antora branches as well. As such, the main aim here was to test this is working.

If we want to allow branch builds for the old jekyll docs then we'll need to come up with a specific branch structure that won't conflict.

**Review by metricjs — APPROVED** [review]

**Review by tiny-ben-tran — APPROVED** [review]

File: .github/workflows/feature_5_legacy_docs.yml (hunk 1 above)
Comment (TheSpyder) [reply .github/workflows/feature_5_legacy_docs.yml:9]: I can think of other ways, but they aren't worth the hassle. We only need the bare minimum until the 5x Antora branch is ready.

File: rtc/jwt-authentication.md [hunk 2]
```diff
@@ -189,7 +189,7 @@ tinymce.init({
       body: JSON.stringify({ documentId }),
     })
     .then((response) => response.json())
-    .then(({ jwt })) => ({ token: jwt })
+    .then(({ jwt }) => ({ token: jwt }))
```
Comment (TheSpyder) [inline rtc/jwt-authentication.md:192]: oh that reminds me I was going to try to clarify this line for a customer. They were returning `{ token }` from their API, didn't realise this was being clever and destructuring expected a `{ jwt }` object.

I'll do it in 6 later.

**Review by TheSpyder — APPROVED** [review]

File: .github/workflows/feature_5_legacy_docs.yml (hunk 1 above)
Comment (lnewson) [reply .github/workflows/feature_5_legacy_docs.yml:9]: Yeah I also had some options (e.g. using the old docker image wercker used) but I also just didn't see the point atm.

Comment (lnewson) [reply .github/workflows/feature_5_legacy_docs.yml:9]: Oops forgot what this comment was about. Ignore the above largely sorry 😞 

Anyways, yeah the only time we might need it is for the 5.10.x patches until the new site is ready, so I'm fine with this manual process for now.

File: rtc/jwt-authentication.md (hunk 2 above)
Comment (TheSpyder) [reply rtc/jwt-authentication.md:192]: Logged as DOC-1641 and I'm about to submit a PR 🤷

**Review by tiny-ben-tran — APPROVED** [review]

**Pull request merged**